File: 2task/Sort/sort.rb
# frozen_string_literal: true

# Sort class
class Sort
  def bubble_sort(array)
    swapped = true
    while swapped
      swapped = false
      array.size - 1.each do |i|
        if array[i] > array[i + 1]
          array[i], array[i + 1] = array[i + 1], array[i]
          swapped = true
        end
      end
    end
  end
end

File: 2task/Sort/sort.cpp
#include <iostream>

void Swap(int* x, int* y)
{
	int temp = *x;
	*x = *y;
	*y = temp;
}

void BubbleSort(int arr[], int length)
{
	bool swapped;

	for (int i = 0; i < length - 1; i++)
	{
		swapped = false;

		for (int j = 0; j < length - i - 1; j++)
		{
			if (arr[j] > arr[j + 1])
			{
				Swap(&arr[j], &arr[j + 1]);
				swapped = true;
			}
		}

		if (!swapped)
			break;
	}
}

int main()
{
	int arr[] = { 2, 1, 8, 7, 5, 6, 3, 4, 9 };
	int length = sizeof(arr) / sizeof(*arr);

	BubbleSort(arr, length);
	for (int i = 0; i < length; i++)
	{
		std::cout << arr[i] << " ";
	}
	std::cout << "\n";

	std::cin.get();
}